contracts/price/ZNSCurvePricer.sol

```diff
@@ -134,20 +142,20 @@ contract ZNSCurvePricer is AAccessControlled, ARegistryWired, UUPSUpgradeable, I
         bytes32 domainHash,
         CurvePriceConfig calldata priceConfig
     ) public override {
-        setPrecisionMultiplier(domainHash, priceConfig.precisionMultiplier);
+        _setPrecisionMultiplier(domainHash, priceConfig.precisionMultiplier);
         priceConfigs[domainHash].baseLength = priceConfig.baseLength;
         priceConfigs[domainHash].maxPrice = priceConfig.maxPrice;
-        priceConfigs[domainHash].minPrice = priceConfig.minPrice;
+        priceConfigs[domainHash].curveMultiplier = priceConfig.curveMultiplier;
         priceConfigs[domainHash].maxLength = priceConfig.maxLength;
         setFeePercentage(domainHash, priceConfig.feePercentage);
```

Comment:
this function will call `validateConfig` on it's own, then it will be called again a few lines below. Do what you did above with `setPrecisionMultiplier`, break it into two functions where the core is internal, and then the public one calls to validate, and then also to the internal function. Then `setPriceConfig` can call to the internal function only and it will avoid reaching `validateConfig` twice.